src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -299,6 +299,8 @@ public Index.Searcher indexSearcher()
      */
     public abstract ClusteringIndexFilter clusteringIndexFilter(DecoratedKey key);
 
+    public abstract DataRange dataRange();
```

Comment:
Have a default implementation throwing USOE so that code doesn't need to be replicated